internal/data/generated_metrics.go

```diff
@@ -95,42 +95,57 @@ func (es ResourceMetricsSlice) Resize(from, to int) {
 // This is a reference type, if passsed by value and callee modifies it the
 // caller will see the modification.
 //
-// Must use NewResourceMetrics function to create new instances.
+// Must use NewEmptyResourceMetrics function to create new instances.
 // Important: zero-initialized instance is not valid for use.
 type ResourceMetrics struct {
 	// Wrap OTLP otlpmetrics.ResourceMetrics.
 	orig *otlpmetrics.ResourceMetrics
 }
 
-// NewResourceMetrics creates a new empty ResourceMetrics.
-func NewResourceMetrics() ResourceMetrics {
-	return ResourceMetrics{&otlpmetrics.ResourceMetrics{}}
-}
-
 func newResourceMetrics(orig *otlpmetrics.ResourceMetrics) ResourceMetrics {
 	return ResourceMetrics{orig}
 }
 
+// NewEmptyResourceMetrics creates a new empty ResourceMetrics.
+//
+// This must be used only in testing code since no "Set" method available.
```

Comment:
I do not quite understand the comment. We have `SetInstrumentationLibraryMetrics`. Does the comment talk about other kind of "Set" method?

Also, if this must be used only in testing code, how should we create a ResourceMetrics in non-testing code? Is it only possible using `NewResourceMetricsSlice`?

Reply:
This is not for Slices (this is something will do later to remove that Set as well), it applies to direct messages only like `SetResource`.

To create a `ResourceMetrics` you actually start from the `MetricData` -> Add a new empty `ResourceMetrics` (again Slices will be updated later) -> then if you need to set a `Resource` you initialize with empty then update properties inside the `Resource`. The idea being that you can initialize a message directly in the MetricData and not set it from outside, you can see `MetricData` as a large `Builder`